src/databricks/labs/remorph/snow/sql_transpiler.py

```diff
@@ -45,8 +83,66 @@ def parse_sql_content(self, sql, file_name):
                     for select in expr.find_all(exp.Select, exp.Join, exp.With, bfs=False):
                         yield self._find_root_tables(select), child
 
+    @staticmethod
+    def safe_parse(statements: str, read: Dialect) -> tuple[list[ParsedExpression], list[str]]:
```

Comment:
Can we make this private as we are not going to expose these outside?

Reply:
@bishwajit-db we have made this public so that is can be reused in future, also it will be good for testing purpose